src/pcm/pcm_hw.c

```diff
@@ -1863,6 +1870,26 @@ int _snd_pcm_hw_open(snd_pcm_t **pcmp, const char *name,
 			rate = val;
 			continue;
 		}
+		if (strcmp(id, "min_rate") == 0) {
```

Comment:
I would add extension to the rate field like:

```
rate [ 44100 48000 ]
```

The parser stores this as compound internally:

```
rate.0 44100
rate.1 48000
```

You may use a loop to parse this or snd_config_search() fcn (and look for '0' and '1' keys).  There is snd_config_is_array() to check, if the compound is array.